src/codegen/src/main/scala/DocGen.scala

```diff
@@ -38,12 +38,14 @@ object DocGen {
 
   def genRstFiles(): Unit = {
     // Generate a modules.rst file that lists all the .py files to be included in API documentation
-    val moduleString = allFiles(toZipDir, _.getName.endsWith(".py"))
-      .map(f => s"   ${f.getName.dropRight(3)}\n").mkString("")
+    val moduleString = allFiles(toZipDir, _.getName.endsWith(".py")).filterNot(_.getName.startsWith("_"))
+        .filterNot(_.getName.equals("java_params_patch.py"))
+        .map(f => s"   ${f.getName.dropRight(3)}\n").mkString("")
     writeFile(new File(inDocDir, "modules.rst"), rstFileLines(moduleString))
 
     // Generate .rst file for each PySpark wrapper - for documentation generation
-    allFiles(toZipDir, _.getName.endsWith(".py"))
+    allFiles(toZipDir, _.getName.endsWith(".py")).filterNot(_.getName.startsWith("_"))
+        .filterNot(_.getName.equals("java_params_patch.py"))
         .foreach{x => writeFile(new File(inDocDir, x.getName.dropRight(3) + ".rst"),
```

Comment:
Minor suggestion: use org.apache.commons.io.FilenameUtils.getBaseName instead of dropRight(3).